<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>DGE clustering with ggplot2 | Daddy&#39;s data analysis by Kazu Nozue</title>
    <link rel="stylesheet" href="/css/style.css" />
    <link rel="stylesheet" href="/css/fonts.css" />
    <link href="//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet">

  </head>

  <body class="page">
    <nav class="header">
      <div class="banner">
<a href="/">

<img src="/post/concept_network2title.png" alt="Daddy&#39;s data analysis for biologist, athletes, and family" />
</a>
</div>

      <div class="head-meta">
      
        <span><a href="/">&larr; Back to Home</a></span>
        <span class="date">2018-10-24</span>
        
        
        
          
        
        
        
        
      
      </div>
    </nav>

<div class="container">
<article>
<div class="article-meta">

  <div class="categories">
  
    <a href="/categories/data-analysis">data-analysis</a>
  
  </div>

  <h1><span class="title">DGE clustering with ggplot2</span></h1>

  
  <h3 class="author">Kazu Nozue
</h3>
  

  
  <p>Tags: <a href="/tags/r">R</a>; <a href="/tags/reproducibility">reproducibility</a>
  </p>
  
  

</div>



<main>
<p>Hierarchical clustering is powerful method to classify gene expression pattern. Here I applied a method develped by <a href="https://plot.ly/ggplot2/ggdendro-dendrograms/">an example of plotly</a>. <!--[test](/post/2018-10-24-dge-clustering-with-ggplot2_files/Screen Shot 2018-10-12 at 3.34.17 PM.png){width=400px}--> # required libraries</p>
<pre class="r"><code># install.packages(&#39;ggdendro&#39;)
library(tidyverse);library(ggdendro);library(plotly)</code></pre>
<pre><code>## ── Attaching packages ──────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>## ✔ ggplot2 3.0.0     ✔ purrr   0.2.5
## ✔ tibble  1.4.2     ✔ dplyr   0.7.7
## ✔ tidyr   0.8.1     ✔ stringr 1.3.1
## ✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>## ── Conflicts ─────────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre><code>## 
## Attaching package: &#39;plotly&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggplot2&#39;:
## 
##     last_plot</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<pre><code>## The following object is masked from &#39;package:graphics&#39;:
## 
##     layout</code></pre>
<div id="readling-gene-lists-and-expression-value" class="section level1">
<h1>readling gene lists and expression value</h1>
<pre class="r"><code># reading gene expression value (Ding, 2018) from my privious blog ([&quot;Differential expression analysis with public available sequencing data&quot;](https://knozue.github.io/post/2018/10/01/differential-expression-analysis-with-public-available-sequencing-data.html) on 2018-10-01))
load(file.path(&quot;2018-10-24-dge-clustering-with-ggplot2_files&quot;,&quot;dge.nolow.Ding2018.Rdata&quot;))
dge.nolow</code></pre>
<pre><code>## Loading required package: edgeR</code></pre>
<pre><code>## Loading required package: limma</code></pre>
<pre><code>## An object of class &quot;DGEList&quot;
## $counts
##   SRR6739807 SRR6739808 SRR6739809 SRR6739810 SRR6739811 SRR6739812
## 1   3598.000   3442.000       3653       3531       4027       3775
## 4    427.000    412.000        316        348        319        357
## 5    132.805    129.168        154        141        155        131
## 6     52.000     46.000         74        110        122         91
## 7    685.000    764.000        231        220        264        247
##   SRR6739813 SRR6739814 SRR6739815 SRR6739816 SRR6739817 SRR6739818
## 1       4195  4317.9900   4015.000   3874.000   3663.000       3642
## 4        286   306.0000    389.000    362.000    290.000        285
## 5        126    99.8598    156.895    130.783    101.738        120
## 6        118   135.0000     60.000     52.000     85.000         79
## 7        118   131.0000    416.000    383.000    104.000        164
## 16794 more rows ...
## 
## $samples
##                     group lib.size norm.factors        Run
## SRR6739807     WT_treated 21747803    0.9813040 SRR6739807
## SRR6739808     WT_treated 21005249    0.9969243 SRR6739808
## SRR6739809   WT_untreated 21052743    0.9901012 SRR6739809
## SRR6739810   WT_untreated 22518666    0.9893923 SRR6739810
## SRR6739811 npr1-1_treated 21437014    1.0185940 SRR6739811
##                      sample        group.1 genotype treatment rep
## SRR6739807     WT_treated_1     WT_treated       WT   treated   1
## SRR6739808     WT_treated_2     WT_treated       WT   treated   2
## SRR6739809   WT_untreated_1   WT_untreated       WT untreated   1
## SRR6739810   WT_untreated_2   WT_untreated       WT untreated   2
## SRR6739811 npr1-1_treated_1 npr1-1_treated   npr1-1   treated   1
##            LibraryLayout
## SRR6739807        SINGLE
## SRR6739808        SINGLE
## SRR6739809        SINGLE
## SRR6739810        SINGLE
## SRR6739811        SINGLE
## 7 more rows ...
## 
## $genes
##         genes
## 1 AT1G50920.1
## 4 AT1G15970.1
## 5 AT1G73440.1
## 6 AT1G75120.1
## 7 AT1G17600.1
## 16794 more rows ...</code></pre>
<pre class="r"><code>cpm.wide &lt;- bind_cols(dge.nolow$gene,as_tibble(cpm(dge.nolow))) %&gt;% as_tibble() %&gt;% rename(transcript_ID=genes)
cpm.wide</code></pre>
<pre><code>## # A tibble: 16,799 x 13
##    transcript_ID SRR6739807 SRR6739808 SRR6739809 SRR6739810 SRR6739811
##    &lt;chr&gt;              &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;
##  1 AT1G50920.1       169.       164.       175.       158.       184.  
##  2 AT1G15970.1        20.0       19.7       15.2       15.6       14.6 
##  3 AT1G73440.1         6.22       6.17       7.39       6.33       7.10
##  4 AT1G75120.1         2.44       2.20       3.55       4.94       5.59
##  5 AT1G17600.1        32.1       36.5       11.1        9.87      12.1 
##  6 AT1G77370.1        34.1       36.3       23.7       25.1       22.9 
##  7 AT1G10950.1       172.       171.       133.       148.       173.  
##  8 AT1G31870.1        40.8       38.5       26.8       26.9       33.3 
##  9 AT1G51360.1         7.40       7.64       9.07      10.4        6.82
## 10 AT1G50950.1         4.45       4.49       3.74       5.16       5.50
## # ... with 16,789 more rows, and 7 more variables: SRR6739812 &lt;dbl&gt;,
## #   SRR6739813 &lt;dbl&gt;, SRR6739814 &lt;dbl&gt;, SRR6739815 &lt;dbl&gt;,
## #   SRR6739816 &lt;dbl&gt;, SRR6739817 &lt;dbl&gt;, SRR6739818 &lt;dbl&gt;</code></pre>
</div>
<div id="dge-list-of-interest" class="section level1">
<h1>DGE list of interest</h1>
<pre class="r"><code># Ding (2018) Cell DEGs
getwd()</code></pre>
<pre><code>## [1] &quot;/Volumes/data_personal/Kazu_blog/content/post&quot;</code></pre>
<pre class="r"><code>DEG.objs&lt;-list.files(path=file.path(&quot;2018-10-24-dge-clustering-with-ggplot2_files&quot;,&quot;output_copy_DEG&quot;),pattern=&quot;\\.csv$&quot;) # under construction
DEG.objs</code></pre>
<pre><code>## [1] &quot;genotypenpr1_genotypenpr1.treatmenttreated.DEGs.int.rWT.rU.csv&quot;
## [2] &quot;genotypenpr4_genotypenpr4.treatmenttreated.DEGs.int.rWT.rU.csv&quot;
## [3] &quot;trt.DEGs.int.rWT.rU.csv&quot;</code></pre>
<pre class="r"><code># read csv file
DEG.list&lt;-lapply(DEG.objs, function(x) read_csv(file.path(&quot;2018-10-24-dge-clustering-with-ggplot2_files&quot;,&quot;output_copy_DEG&quot;,x)))</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   genes = col_character(),
##   logFC.genotypenpr1.1 = col_double(),
##   logFC.genotypenpr1.1.treatmenttreated = col_double(),
##   logCPM = col_double(),
##   LR = col_double(),
##   PValue = col_double(),
##   FDR = col_double(),
##   AGI = col_character(),
##   symbol2 = col_character(),
##   full_name = col_character()
## )</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   genes = col_character(),
##   logFC.genotypenpr4.4D = col_double(),
##   logFC.genotypenpr4.4D.treatmenttreated = col_double(),
##   logCPM = col_double(),
##   LR = col_double(),
##   PValue = col_double(),
##   FDR = col_double(),
##   AGI = col_character(),
##   symbol2 = col_character(),
##   full_name = col_character()
## )</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   genes = col_character(),
##   logFC = col_double(),
##   logCPM = col_double(),
##   LR = col_double(),
##   PValue = col_double(),
##   FDR = col_double(),
##   AGI = col_character(),
##   symbol2 = col_character(),
##   full_name = col_character()
## )</code></pre>
<pre class="r"><code>names(DEG.list)&lt;-gsub(&quot;.csv&quot;,&quot;&quot;,DEG.objs)
head(DEG.list)</code></pre>
<pre><code>## $genotypenpr1_genotypenpr1.treatmenttreated.DEGs.int.rWT.rU
## # A tibble: 6,460 x 10
##    genes logFC.genotypen… logFC.genotypen… logCPM    LR    PValue       FDR
##    &lt;chr&gt;            &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##  1 AT2G…           -4.19            -0.733   6.69 1327. 6.75e-289 1.13e-284
##  2 AT5G…            3.26             0.517   4.76 1072. 1.33e-233 1.11e-229
##  3 AT1G…           -2.01            -0.488   7.19  659. 6.38e-144 3.57e-140
##  4 AT1G…           -2.25            -0.525   7.51  624. 2.54e-136 1.07e-132
##  5 AT3G…           -3.15             0.874   5.33  622. 1.00e-135 3.36e-132
##  6 AT5G…           -0.206           -6.73    3.90  595. 7.88e-130 2.21e-126
##  7 AT1G…           -3.64            -1.53    3.37  581. 7.28e-127 1.75e-123
##  8 AT5G…            1.83             1.82    9.40  571. 1.04e-124 2.18e-121
##  9 AT2G…            2.23             1.94    9.22  559. 4.26e-122 7.94e-119
## 10 AT5G…           -2.01            -3.49    4.19  538. 1.19e-117 2.00e-114
## # ... with 6,450 more rows, and 3 more variables: AGI &lt;chr&gt;,
## #   symbol2 &lt;chr&gt;, full_name &lt;chr&gt;
## 
## $genotypenpr4_genotypenpr4.treatmenttreated.DEGs.int.rWT.rU
## # A tibble: 854 x 10
##    genes logFC.genotypen… logFC.genotypen… logCPM    LR    PValue       FDR
##    &lt;chr&gt;            &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##  1 AT5G…             3.04            0.632   4.76  966. 1.54e-210 2.59e-206
##  2 AT5G…            -3.31            0.358   3.10  463. 3.40e-101 2.85e- 97
##  3 AT3G…            -5.46           -0.178   1.98  454. 2.07e- 99 1.16e- 95
##  4 AT1G…            -7.41           -3.70    2.00  424. 6.68e- 93 2.80e- 89
##  5 AT3G…            -2.97           -0.297   3.36  393. 4.39e- 86 1.47e- 82
##  6 AT2G…            -1.42           -0.699   6.69  284. 2.63e- 62 7.35e- 59
##  7 AT1G…            -2.50            0.894   6.30  280. 1.23e- 61 2.96e- 58
##  8 AT2G…            -1.10           -0.233   5.87  221. 9.62e- 49 2.02e- 45
##  9 AT5G…            -3.04            0.438   4.19  208. 6.15e- 46 1.15e- 42
## 10 AT5G…            -1.84           -0.485   2.78  190. 4.64e- 42 7.79e- 39
## # ... with 844 more rows, and 3 more variables: AGI &lt;chr&gt;, symbol2 &lt;chr&gt;,
## #   full_name &lt;chr&gt;
## 
## $trt.DEGs.int.rWT.rU
## # A tibble: 7,862 x 9
##    genes  logFC logCPM    LR    PValue       FDR AGI   symbol2 full_name  
##    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;      
##  1 AT5G1…  8.58   3.90  662. 5.49e-146 9.22e-142 AT5G… AT-HSP… heat shock…
##  2 AT1G0…  4.01   4.60  613. 2.14e-135 1.80e-131 AT1G… &lt;NA&gt;    &lt;NA&gt;       
##  3 AT4G2…  3.86   5.39  586. 2.36e-129 1.32e-125 AT4G… CRK14   cysteine-r…
##  4 AT5G2…  6.17   4.19  582. 1.27e-128 5.32e-125 AT5G… ATWRKY… ARABIDOPSI…
##  5 AT3G2…  4.93   3.36  578. 9.46e-128 3.18e-124 AT3G… &lt;NA&gt;    &lt;NA&gt;       
##  6 AT5G1…  9.65   3.77  569. 1.18e-125 3.30e-122 AT5G… HSP17.… 17.6 kDa c…
##  7 AT4G2…  3.34   6.70  563. 1.80e-124 4.32e-121 AT4G… SGT1A   &lt;NA&gt;       
##  8 AT1G2…  2.88   7.19  539. 2.54e-119 5.34e-116 AT1G… AtWAK1… NA;NA;cell…
##  9 AT5G2…  3.83   6.70  529. 4.88e-117 9.12e-114 AT5G… AtDMR6… NA;DOWNY M…
## 10 AT2G1…  2.96   7.30  512. 2.22e-113 3.73e-110 AT2G… ATSERK… SOMATIC EM…
## # ... with 7,852 more rows</code></pre>
</div>
<div id="making-matrix-for-degs" class="section level1">
<h1>making matrix for DEGs</h1>
<pre class="r"><code>DEG.genes&lt;-unique(as_vector(rbind(DEG.list[[1]][DEG.list[[1]]$FDR&lt;1e-30,&quot;genes&quot;],
                                DEG.list[[2]][DEG.list[[2]]$FDR&lt;1e-30,&quot;genes&quot;],
                                DEG.list[[3]][DEG.list[[3]]$FDR&lt;1e-30,&quot;genes&quot;])))
DEG.DF&lt;- cpm.wide %&gt;% filter(transcript_ID %in% DEG.genes)
dim(DEG.DF) # 305 13</code></pre>
</div>
<div id="an-example" class="section level1">
<h1>An example</h1>
<pre class="r"><code>#dendogram data
x &lt;- as.matrix(scale(t(DEG.DF[,-1])))</code></pre>
<pre><code>## Error in t(DEG.DF[, -1]): object &#39;DEG.DF&#39; not found</code></pre>
<pre class="r"><code>dd.col &lt;- as.dendrogram(hclust(dist(x)))</code></pre>
<pre><code>## Error in as.matrix(x): object &#39;x&#39; not found</code></pre>
<pre class="r"><code>dd.row &lt;- as.dendrogram(hclust(dist(t(x))))</code></pre>
<pre><code>## Error in t(x): object &#39;x&#39; not found</code></pre>
<pre class="r"><code>dx &lt;- dendro_data(dd.row)</code></pre>
<pre><code>## Error in dendro_data(dd.row): object &#39;dd.row&#39; not found</code></pre>
<pre class="r"><code>dy &lt;- dendro_data(dd.col)</code></pre>
<pre><code>## Error in dendro_data(dd.col): object &#39;dd.col&#39; not found</code></pre>
<pre class="r"><code># helper function for creating dendograms
ggdend &lt;- function(df) {
  ggplot() +
    geom_segment(data = df, aes(x=x, y=y, xend=xend, yend=yend)) +
    labs(x = &quot;&quot;, y = &quot;&quot;) + theme_minimal() +
    theme(axis.text = element_blank(), axis.ticks = element_blank(),
          panel.grid = element_blank())
}

# x/y dendograms
px &lt;- ggdend(dx$segments)</code></pre>
<pre><code>## Error in fortify(data): object &#39;dx&#39; not found</code></pre>
<pre class="r"><code>py &lt;- ggdend(dy$segments) + coord_flip()</code></pre>
<pre><code>## Error in fortify(data): object &#39;dy&#39; not found</code></pre>
<pre class="r"><code># heatmap
col.ord &lt;- order.dendrogram(dd.col)</code></pre>
<pre><code>## Error in order.dendrogram(dd.col): object &#39;dd.col&#39; not found</code></pre>
<pre class="r"><code>row.ord &lt;- order.dendrogram(dd.row)</code></pre>
<pre><code>## Error in order.dendrogram(dd.row): object &#39;dd.row&#39; not found</code></pre>
<pre class="r"><code>df &lt;- as.data.frame(x)</code></pre>
<pre><code>## Error in as.data.frame(x): object &#39;x&#39; not found</code></pre>
<pre class="r"><code>rowSums(df);colSums(df)</code></pre>
<pre><code>## Error in rowSums(df): &#39;x&#39; must be an array of at least two dimensions</code></pre>
<pre><code>## Error in colSums(df): &#39;x&#39; must be an array of at least two dimensions</code></pre>
<pre class="r"><code>#xx_names &lt;- attr(xx, &quot;dimnames&quot; &lt;- as.data.frame(xx)
##### under construction #########

colnames(df) &lt;- factor(as_vector(DEG.DF[,1]),levels=as_vector(DEG.DF[,1])[row.ord])</code></pre>
<pre><code>## Error in vapply(x, is.atomic, logical(1)): object &#39;DEG.DF&#39; not found</code></pre>
<pre class="r"><code>#rownames(df) &lt;- factor(rownames(DEG.DF),levels=rownames(DEG.DF)[row.ord])

#df$car &lt;- xx_names[[1]]
#df$car &lt;- with(df, factor(car, levels=car, ordered=TRUE))
mdf &lt;- as.data.frame(t(df)) %&gt;% rownames_to_column(var=&quot;transcript_ID&quot;) %&gt;% gather(key=&quot;sample&quot;,value=&quot;value&quot;,-1)</code></pre>
<pre><code>## Error in t.default(df): argument is not a matrix</code></pre>
<pre class="r"><code>mdf$transcript_ID&lt;-factor(mdf$transcript_ID,levels=as_vector(DEG.DF[,1])[row.ord])</code></pre>
<pre><code>## Error in factor(mdf$transcript_ID, levels = as_vector(DEG.DF[, 1])[row.ord]): object &#39;mdf&#39; not found</code></pre>
<pre class="r"><code>p &lt;- ggplot(mdf, aes(x = sample, y = transcript_ID)) + geom_tile(aes(fill = value))</code></pre>
<pre><code>## Error in ggplot(mdf, aes(x = sample, y = transcript_ID)): object &#39;mdf&#39; not found</code></pre>
<pre class="r"><code>p</code></pre>
<pre><code>## Error in eval(expr, envir, enclos): object &#39;p&#39; not found</code></pre>
<pre class="r"><code># hide axis ticks and grid lines
eaxis &lt;- list(
  showticklabels = FALSE,
  showgrid = FALSE,
  zeroline = FALSE
)

p_empty &lt;- plot_ly(filename=&quot;dendrogram&quot;) %&gt;%
  # note that margin applies to entire plot, so we can
  # add it here to make tick labels more readable
  layout(margin = list(l = 200),
         xaxis = eaxis,
         yaxis = eaxis)</code></pre>
<pre><code>## Warning in plot_ly(filename = &quot;dendrogram&quot;): Ignoring filename. Use
## `plotly_POST()` if you want to post figures to plotly.</code></pre>
<pre class="r"><code>Sys.setenv(&quot;plotly_username&quot;=&quot;knozue&quot;)
Sys.setenv(&quot;plotly_api_key&quot; = &quot;KKbg9Oe9ETsNXMlPZOix&quot;)
#signup(username=&quot;knozue&quot;,email=&quot;kazunozue@gmail.com&quot;)
#signup(&#39;knozue&#39;,api_key=&quot;KKbg9Oe9ETsNXMlPZOix&quot;)
#signup(email=&#39;kazunozue@gmail.com&#39;,api_key=&quot;KKbg9Oe9ETsNXMlPZOix&quot;)
q&lt;-subplot(px, p_empty, p, py, nrows = 2, margin = 0.01)</code></pre>
<pre><code>## Error in dots2plots(...): object &#39;px&#39; not found</code></pre>
<pre class="r"><code>q</code></pre>
<pre><code>## function (save = &quot;default&quot;, status = 0, runLast = TRUE) 
## .Internal(quit(save, status, runLast))
## &lt;bytecode: 0x7fd6f2980850&gt;
## &lt;environment: namespace:base&gt;</code></pre>
<p>To save image, you need to sign-in <a href="https://plot.ly/">plotly website</a> and set environmental variables as described in signup() function help. And then run one line below</p>
<pre class="r"><code>plotly_IMAGE(q,width=1000,height=500, format=&quot;png&quot;,scale=1, out_file=&quot;dendrogram2.png&quot;)</code></pre>
</div>
<div id="more-advanced-methods" class="section level1">
<h1>More advanced methods</h1>
<p>dendextend package <a href="https://www.ncbi.nlm.nih.gov/pubmed/26209431">Tal Galili (2015)</a>, whose updated version can be found in <a href="https://github.com/talgalili/dendextend/">dendextend github</a>.</p>
</div>
<div id="references" class="section level1">
<h1>references</h1>
<p><span class="citation">@Article</span>{, author = {Tal Galili}, title = {dendextend: an R package for visualizing, adjusting, and comparing trees of hierarchical clustering}, journal = {Bioinformatics}, year = {2015}, doi = {10.1093/bioinformatics/btv428}, url = {<a href="https://academic.oup.com/bioinformatics/article/31/22/3718/240978/dendextend-an-R-package-for-visualizing-adjusting" class="uri">https://academic.oup.com/bioinformatics/article/31/22/3718/240978/dendextend-an-R-package-for-visualizing-adjusting</a>}, eprint = {<a href="https://academic.oup.com/bioinformatics/article-pdf/31/22/3718/17122682/btv428.pdf" class="uri">https://academic.oup.com/bioinformatics/article-pdf/31/22/3718/17122682/btv428.pdf</a>}, }</p>
</div>

</main>














<nav class="post-nav">
  <span class="nav-prev"><a href="/post/2018/10/26/another-plant-defense-hormone-is-required-for-responses-to-light-in-plant.html">&larr; Another plant defense hormone is required for responses to light in plant</a></span>
  <span class="nav-next"><a href="/post/2018/10/18/native-plant-garden-project.html">Native plant garden project &rarr;</a></span>
</nav>



</article>
</div>

<script async src="//yihui.name/js/center-img.js"></script>

<footer>




<link href="//cdn-images.mailchimp.com/embedcode/horizontal-slim-10_7.css" rel="stylesheet" type="text/css">
<style type="text/css">
	#mc_embed_signup{background:#fff; clear:left; font:14px Helvetica,Arial,sans-serif; width:100%;}
	 
</style>
<div id="mc_embed_signup">
<form action="https://github.us19.list-manage.com/subscribe/post?u=6998dd4b198ff6953fa7c4c57&amp;id=e3202ca383" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
    <div id="mc_embed_signup_scroll">
	<label for="mce-EMAIL">Subscribe to my mailing list (free)</label>
	<input type="email" value="" name="EMAIL" class="email" id="mce-EMAIL" placeholder="email address" required>
    
    <div style="position: absolute; left: -5000px;" aria-hidden="true"><input type="text" name="b_6998dd4b198ff6953fa7c4c57_e3202ca383" tabindex="-1" value=""></div>
    <div class="clear"><input type="submit" value="Subscribe" name="subscribe" id="mc-embedded-subscribe" class="button"></div>
    </div>
</form>
</div>




<div class="footer">
  <ul class="menu">
    
    <li><a href="/"><span data-hover="Home">Home</span></a></li>
    
    <li><a href="/about/"><span data-hover="About">About</span></a></li>
    
    <li><a href="/categories/"><span data-hover="Categories">Categories</span></a></li>
    
    <li><a href="/tags/"><span data-hover="Tags">Tags</span></a></li>
    
    <li><a href="/post/"><span data-hover="Post">Post</span></a></li>
    
    <li><a href="/index.xml"><span data-hover="RSS feed">RSS feed</span></a></li>
    
  </ul>
  
</div>
</footer>


<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script>
hljs.configure({languages: []});
hljs.initHighlightingOnLoad();
</script>



<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-96166887-2', 'auto');
	
	ga('send', 'pageview');
}
</script>


</body>
</html>

