<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>Over-representation analysis 3: GOseq with GO term in non-model organisms | Daddy&#39;s data analysis by Kazu Nozue</title>
    <link rel="stylesheet" href="/css/style.css" />
    <link rel="stylesheet" href="/css/fonts.css" />
    <link href="//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet">

  </head>

  <body class="page">
    <nav class="header">
      <div class="banner">
<a href="/">

<img src="/post/concept_network2title.png" alt="Daddy&#39;s data analysis for biologist, athletes, and family" />
</a>
</div>

      <div class="head-meta">
      
        <span><a href="/">&larr; Back to Home</a></span>
        <span class="date">2018-10-17</span>
        
        
        
          
        
        
        
        
      
      </div>
    </nav>

<div class="container">
<article>
<div class="article-meta">

  <div class="categories">
  
    <a href="/categories/data-analysis">data-analysis</a>
  
  </div>

  <h1><span class="title">Over-representation analysis 3: GOseq with GO term in non-model organisms</span></h1>

  
  <h3 class="author">Kazu Nozue
</h3>
  

  
  <p>Tags: <a href="/tags/r">R</a>; <a href="/tags/reproducibility">reproducibility</a>
  </p>
  
  

</div>



<main>
<div id="if-you-want-to-analyze-non-model-organisms-what-you-can-do" class="section level1">
<h1>If you want to analyze non-model organisms, what you can do?</h1>
<p>Please prepare</p>
<ol class="example" style="list-style-type: decimal">
<li>fasta file for cDNA of your species (eg. Brassica rapa),</li>
<li>fasta file for cDNA of a close model organisms to your species (eg. Arabidopsis thaliana for Brassica rapa),</li>
<li>blastn program installed in your computer (UNIX (or Mac OSX) or LINUX) (<a href="https://www.ncbi.nlm.nih.gov/books/NBK52640/">detailed instruciton from NCBI</a>) <!--[](/post/2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Trello_example_Screen Shot 2018-09-20 at 9.58.20 AM.png)--></li>
</ol>
</div>
<div id="required-libraries" class="section level1">
<h1>required libraries</h1>
<pre class="r"><code>library(tidyverse);library(readr);library(readxl)</code></pre>
<pre><code>## ── Attaching packages ───────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>## ✔ ggplot2 3.0.0     ✔ purrr   0.2.5
## ✔ tibble  1.4.2     ✔ dplyr   0.7.7
## ✔ tidyr   0.8.1     ✔ stringr 1.3.1
## ✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>## ── Conflicts ──────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
</div>
<div id="blastn" class="section level1">
<h1>BLASTN</h1>
<p>Use the <a href="http://brassicadb.org/brad/datasets/pub/Genomes/Brassica_rapa/V1.0/">Chifu V1.0 file</a> BLAST against <a href="ftp://ftp.arabidopsis.org/home/tair/Sequences/blast_datasets/TAIR10_blastsets/TAIR10_cds_20110103_representative_gene_model_updated">TAIR10 CDS</a></p>
<p>Because these organisms are relatively closely related I will use blastn instead of blastp to focus on best match</p>
<pre class="r"><code># look Julin&#39;s scripts and work by myself
list.files()
setwd(&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files&quot;)
system(&quot;makeblastdb -in TAIR10_cds_20110103_representative_gene_model_updated.fa -dbtype nucl&quot;)
#
system(&quot;gunzip Brassica_rapa.20100830.cds.gz&quot;)
# BLASTN
system(&quot;blastn -query Brassica_rapa.20100830.cds -db TAIR10_cds_20110103_representative_gene_model_updated.fa -strand both -task dc-megablast -outfmt 10 -culling_limit 1 -max_hsps 1 -evalue 10e-4 -num_threads 3 -template_type coding -template_length 16 -out Brapa1.0_vs_At_dc-megablast_out.csv&quot;)
# system(&quot;rm Brassica_rapa.20100830.cds&quot;)
setwd(&quot;..&quot;)</code></pre>
</div>
<div id="import-results" class="section level1">
<h1>import results</h1>
<pre class="r"><code>brapa.blast &lt;- read_csv(&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Brapa1.0_vs_At_dc-megablast_out.csv&quot;, col_names = FALSE)
colnames(brapa.blast) &lt;- c(&quot;query&quot;,&quot;subject&quot;,&quot;perc_ID&quot;,&quot;aln_length&quot;,&quot;mismatch&quot;,&quot;gap_open&quot;,&quot;qstart&quot;,&quot;qend&quot;,&quot;sstart&quot;,&quot;send&quot;,&quot;eval&quot;,&quot;score&quot;)
head(brapa.blast)
summary(brapa.blast)
brapa.blast %&gt;% group_by(query) %&gt;% summarise(n())
# select highest score within one query
brapa.blast %&gt;% group_by(query) %&gt;% filter(score==max(score)) %&gt;% summarise(n())
Br.v1.0anno.At.BLAST &lt;- brapa.blast %&gt;% group_by(query) %&gt;% filter(score==max(score))
write_csv(Br.v1.0anno.At.BLAST,path=&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Brapa_V1.0_annotated.csv&quot;)</code></pre>
</div>
<div id="reading-br-at-gene-name-conversion-table" class="section level1">
<h1>Reading Br-At gene name conversion table</h1>
<pre class="r"><code>### prerequisit: results from reciprocal BLAST against Arabidopsis thaliana
Br.v1.0anno.At.BLAST&lt;-readr::read_csv(file=&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Brapa_V1.0_annotated.csv&quot;) # needs to use updated one from Julin (111718)</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   query = col_character(),
##   subject = col_character(),
##   perc_ID = col_double(),
##   aln_length = col_integer(),
##   mismatch = col_integer(),
##   gap_open = col_integer(),
##   qstart = col_integer(),
##   qend = col_integer(),
##   sstart = col_integer(),
##   send = col_integer(),
##   eval = col_double(),
##   score = col_double()
## )</code></pre>
<pre class="r"><code>Br.v1.0anno.At.BLAST</code></pre>
<pre><code>## # A tibble: 37,670 x 12
##    query subject perc_ID aln_length mismatch gap_open qstart  qend sstart
##    &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;      &lt;int&gt;    &lt;int&gt;    &lt;int&gt;  &lt;int&gt; &lt;int&gt;  &lt;int&gt;
##  1 Bra0… AT2G37…    85.0       1509      149        7      1  1506      1
##  2 Bra0… AT2G37…    86.8       1080      141        2      1  1079      1
##  3 Bra0… AT2G37…    83.1        590       75        7      1   570      1
##  4 Bra0… AT2G37…    81.9        365       63        2      1   365      1
##  5 Bra0… AT2G37…    82.8       1218      161        9      1  1179      1
##  6 Bra0… AT3G53…    85.1        308       46        0      1   308      1
##  7 Bra0… AT2G37…    78.6        729      117       10      1   711      1
##  8 Bra0… AT2G37…    81.7        750       77        7      1   702      1
##  9 Bra0… AT2G37…    88.1        337       37        1    319   652      1
## 10 Bra0… AT2G37…    91.5       1132       96        0      1  1132      1
## # ... with 37,660 more rows, and 3 more variables: send &lt;int&gt;, eval &lt;dbl&gt;,
## #   score &lt;dbl&gt;</code></pre>
</div>
<div id="making-brgo.v1.5anno.atgoslim.bp.list" class="section level1">
<h1>Making Brgo.v1.5anno.Atgoslim.BP.list</h1>
<pre class="r"><code># Reading Arabidopsis thaliana GOslim list object (see [&quot;Over-representation analysis 1: GOseq with GO terms in unsupported model organisms&quot;](https://knozue.github.io/post/2018/09/26/over-representation-analysis-1-goseq-with-arabidopsis-go-term.html) for details)
load(&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Atgoslim.TAIR.BP.list.Rdata&quot;)
head(Atgoslim.TAIR.BP.list)
# remove splicing variant
Br.v1.0anno.At.BLAST &lt;- Br.v1.0anno.At.BLAST %&gt;% mutate(AGI=str_remove(subject,pattern=&quot;\\.[[:digit:]]+&quot;))
# asign At GO into corresponding Br genes
Brgo.v1.0anno.Atgoslim.BP.list&lt;-list()
for(i in 1:length(Br.v1.0anno.At.BLAST$query)) {
    if(is.null(Atgoslim.TAIR.BP.list[[as_vector(Br.v1.0anno.At.BLAST[i,&quot;AGI&quot;])]])) next else {
      Brgo.v1.0anno.Atgoslim.BP.list[[i]]&lt;-Atgoslim.TAIR.BP.list[[as_vector(Br.v1.0anno.At.BLAST[i,&quot;AGI&quot;])]]
      names(Brgo.v1.0anno.Atgoslim.BP.list)[[i]]&lt;-as_vector(Br.v1.0anno.At.BLAST[i,&quot;query&quot;])
    }
  }
  table(sapply(Brgo.v1.0anno.Atgoslim.BP.list,is.null)) # FALSE 33474 TRUE 4196
  # remove gene with no GO term
  Brgo.v1.0anno.Atgoslim.BP.list&lt;-Brgo.v1.0anno.Atgoslim.BP.list[!sapply(Brgo.v1.0anno.Atgoslim.BP.list,is.null)]
  table(sapply(Brgo.v1.0anno.Atgoslim.BP.list,is.null))
  save(Brgo.v1.0anno.Atgoslim.BP.list,file=file.path(&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files&quot;,&quot;Brgo.v1.0anno.Atgoslim.BP.list.Rdata&quot;))
#} else {print(&quot;change directory&quot;);stop}</code></pre>
</div>
<div id="prep-for-brassica-rapa-goseq-analysis-v1.0-annotation-for-greenham.2016.clock.drought-data-below" class="section level1">
<h1>prep for Brassica rapa GOseq analysis (v1.0 annotation) (for Greenham.2016.clock.drought data below)</h1>
<pre class="r"><code>load(&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Brgo.v1.0anno.Atgoslim.BP.list.Rdata&quot;) 
library(ShortRead);library(goseq);library(GO.db);library(&quot;annotate&quot;)
# read cDNA fasta file with v1.0 annotation
#Br_cdna&lt;-readDNAStringSet(&quot;http://brassicadb.org/brad/datasets/pub/Genomes/Brassica_rapa/V1.0/Scaffold1.0/Brassica_rapa.20100830.cds.gz&quot;) 
Br_cdna&lt;-readDNAStringSet(&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Brassica_rapa.20100830.cds.gz&quot;) # when above line is slow use this line.
head(Br_cdna)</code></pre>
<pre><code>##   A DNAStringSet instance of length 6
##     width seq                                          names               
## [1]  1515 ATGGGGAAGATCTTGAAAACT...TATCGTCTGCTATTCATTAG Bra000001  [mRNA]...
## [2]  1089 ATGGAGGAAGTAAGGAAGATG...ATGGTGTAGAAACACTCTAA Bra000002  [mRNA]...
## [3]   570 ATGAGCTCTGTTTGTGGTAAG...TAGTTGCAGCATCTTCTTGA Bra000003  [mRNA]...
## [4]   375 ATGATTCGCCGTCTATTCTCG...AAGAAGAAGCAGCCCTTTAA Bra000004  [mRNA]...
## [5]  1434 ATGGCGGCAGCTAGACGATTG...CCGGTGGAGGTTTTCTCTAG Bra000005  [mRNA]...
## [6]   312 ATGTCTGGGCGAGGAAAAGGA...TATATGGATTCGGCGGTTGA Bra000006  [mRNA]...</code></pre>
</div>
<div id="goseq-ora-function-for-brassica-rapa" class="section level1">
<h1>GOseq ORA function for Brassica rapa</h1>
<pre class="r"><code>GOseq.Brgo.v1.0.Atgoslim.BP.list.ORA&lt;-function(genelist,padjust=0.05,ontology=&quot;BP&quot;,custom.category.list=Brgo.v1.0anno.Atgoslim.BP.list,cdna=Br_cdna) { # return GO enrichment table, padjus, padjust=0.05. 
  bias&lt;-nchar(cdna)
  names(bias)&lt;-tibble(Bra=names(cdna)) %&gt;% separate(Bra,into=&quot;Bra2&quot;,sep=&quot; &quot;,extra=&quot;drop&quot;) %&gt;% dplyr::select(Bra2) %&gt;% as_vector()
  TF&lt;-as.integer(names(bias) %in% genelist)
    names(TF)&lt;-names(bias)
    #print(TF)
  pwf&lt;-nullp(TF,bias.data=bias)
  #print(pwf$DEgenes)
  GO.pval &lt;- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #GO.pval &lt;- goseq(pwf,gene2cat=Brgo.DF3,use_genes_without_cat=TRUE) # format became different in new goseq version (021111)
  
  #head(GO.pval) 
  if(ontology==&quot;BP&quot;) {
    GO.pval2&lt;-subset(GO.pval,ontology==&quot;BP&quot;)
  } else if(ontology==&quot;CC&quot;) {
    GO.pval2&lt;-subset(GO.pval,ontology==&quot;CC&quot;)
  } else {
    GO.pval2&lt;-subset(GO.pval,ontology==&quot;MF&quot;)
  }
    
  GO.pval2$over_represented_padjust&lt;-p.adjust(GO.pval2$over_represented_pvalue,method=&quot;BH&quot;)
  if(GO.pval2$over_represented_padjust[1]&gt;padjust) return(&quot;no enriched GO&quot;)
  else {
    enriched.GO&lt;-GO.pval2[GO.pval2$over_represented_padjust&lt;padjust,] 
    print(&quot;enriched.GO is&quot;)
    print(enriched.GO)
    
    ## write Term and Definition 
    for(i in 1:dim(enriched.GO)[1]) {
      if(is.null(Term(GOTERM[enriched.GO[i,&quot;category&quot;]]))) {next} else {
      enriched.GO$Term[i]&lt;-Term(GOTERM[[enriched.GO[i,&quot;category&quot;]]])
      enriched.GO$Definition[i]&lt;-Definition(GOTERM[[enriched.GO[i,&quot;category&quot;]]])
      }
    }
    return(enriched.GO)
  }
}</code></pre>
</div>
<div id="goseq-analysis-of-differentially-expressed-gene-list" class="section level1">
<h1>GOseq analysis of differentially expressed gene list</h1>
<p>Using a Brassica rapa gene list in <a href="https://elifesciences.org/articles/29655">Greenham (2016)</a>. Compare our GOseq results with one in Figure 8B.</p>
<pre class="r"><code># Note: v1.0
download.file(&quot;https://elifesciences.org/download/aHR0cHM6Ly9jZG4uZWxpZmVzY2llbmNlcy5vcmcvYXJ0aWNsZXMvMjk2NTUvZWxpZmUtMjk2NTUtc3VwcDEtdjIueGxzeA==/elife-29655-supp1-v2.xlsx?_hash=IzWtqJP3ae%2BeIAz0DN%2Fyp44pKc2grNAxM7z%2Bd%2FzBibM%3D&quot;,destfile=&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/elife-29655-supp1-v2.xlsx&quot;)
Greenham.2016.clock.drought&lt;-read_excel(path=&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/elife-29655-supp1-v2.xlsx&quot;,sheet=2,skip=4)
system(&quot;rm 2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/elife-29655-supp1-v2.xlsx&quot;)
Greenham.2016.clock.drought.dM5&lt;-Greenham.2016.clock.drought %&gt;% filter(Module==&quot;M5&quot;) %&gt;% dplyr::select(Gene) %&gt;% as_vector() # v1.0 annotation
head(Greenham.2016.clock.drought.dM5)</code></pre>
<pre><code>##       Gene1       Gene2       Gene3       Gene4       Gene5       Gene6 
## &quot;Bra011755&quot; &quot;Bra011741&quot; &quot;Bra011671&quot; &quot;Bra011655&quot; &quot;Bra011606&quot; &quot;Bra011581&quot;</code></pre>
<pre class="r"><code>eGO.dM5&lt;-GOseq.Brgo.v1.0.Atgoslim.BP.list.ORA(genelist=Greenham.2016.clock.drought.dM5)</code></pre>
<pre><code>## Warning in pcls(G): initial point very close to some inequality constraints</code></pre>
<pre><code>## Using manually entered categories.</code></pre>
<pre><code>## Calculating the p-values...</code></pre>
<pre><code>## &#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<p><img src="/post/2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre><code>## [1] &quot;enriched.GO is&quot;
##        category over_represented_pvalue under_represented_pvalue
## 1517 GO:0015979            2.192366e-19                1.0000000
## 977  GO:0009768            2.651766e-17                1.0000000
## 978  GO:0009769            1.204903e-10                1.0000000
## 914  GO:0009645            9.683773e-10                1.0000000
## 913  GO:0009644            3.394233e-08                1.0000000
## 860  GO:0009416            8.568277e-08                1.0000000
## 1168 GO:0010114            2.127734e-07                1.0000000
## 854  GO:0009409            5.068985e-06                0.9999983
## 1720 GO:0019684            8.198031e-06                0.9999997
## 640  GO:0006970            2.153608e-05                0.9999952
##      numDEInCat numInCat
## 1517         27      192
## 977          14       37
## 978           7       13
## 914           7       16
## 913          12      106
## 860          22      386
## 1168         11      100
## 854          26      667
## 1720          5       21
## 640          14      251
##                                                    term ontology
## 1517                                     photosynthesis       BP
## 977   photosynthesis, light harvesting in photosystem I       BP
## 978  photosynthesis, light harvesting in photosystem II       BP
## 914            response to low light intensity stimulus       BP
## 913                    response to high light intensity       BP
## 860                          response to light stimulus       BP
## 1168                              response to red light       BP
## 854                                    response to cold       BP
## 1720                     photosynthesis, light reaction       BP
## 640                          response to osmotic stress       BP
##      over_represented_padjust
## 1517             8.162180e-16
## 977              4.936262e-14
## 978              1.495284e-07
## 914              9.013172e-07
## 913              2.527346e-05
## 860              5.316616e-05
## 1168             1.131651e-04
## 854              2.358979e-03
## 1720             3.391252e-03
## 640              8.017883e-03</code></pre>
<!--KazuRbook1 https://affiliate-program.amazon.com/home/ads/adcode/saved-->
<div id="amzn-assoc-ad-99af12ae-45ba-4556-8dc6-e84f97060807">

</div>
<script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&adInstanceId=99af12ae-45ba-4556-8dc6-e84f97060807"></script>
</div>

</main>














<nav class="post-nav">
  <span class="nav-prev"><a href="/post/2018/10/18/native-plant-garden-project.html">&larr; Native plant garden project</a></span>
  <span class="nav-next"><a href="/post/2018/10/12/merging-two-branches-in-git.html">Merging two branches in Git  &rarr;</a></span>
</nav>



</article>
</div>

<script async src="//yihui.name/js/center-img.js"></script>

<footer>




<link href="//cdn-images.mailchimp.com/embedcode/horizontal-slim-10_7.css" rel="stylesheet" type="text/css">
<style type="text/css">
	#mc_embed_signup{background:#fff; clear:left; font:14px Helvetica,Arial,sans-serif; width:100%;}
	 
</style>
<div id="mc_embed_signup">
<form action="https://github.us19.list-manage.com/subscribe/post?u=6998dd4b198ff6953fa7c4c57&amp;id=e3202ca383" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
    <div id="mc_embed_signup_scroll">
	<label for="mce-EMAIL">Subscribe to my mailing list (free)</label>
	<input type="email" value="" name="EMAIL" class="email" id="mce-EMAIL" placeholder="email address" required>
    
    <div style="position: absolute; left: -5000px;" aria-hidden="true"><input type="text" name="b_6998dd4b198ff6953fa7c4c57_e3202ca383" tabindex="-1" value=""></div>
    <div class="clear"><input type="submit" value="Subscribe" name="subscribe" id="mc-embedded-subscribe" class="button"></div>
    </div>
</form>
</div>




<div class="footer">
  <ul class="menu">
    
    <li><a href="/"><span data-hover="Home">Home</span></a></li>
    
    <li><a href="/about/"><span data-hover="About">About</span></a></li>
    
    <li><a href="/categories/"><span data-hover="Categories">Categories</span></a></li>
    
    <li><a href="/tags/"><span data-hover="Tags">Tags</span></a></li>
    
    <li><a href="/post/"><span data-hover="Post">Post</span></a></li>
    
    <li><a href="/index.xml"><span data-hover="RSS feed">RSS feed</span></a></li>
    
  </ul>
  
</div>
</footer>


<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script>
hljs.configure({languages: []});
hljs.initHighlightingOnLoad();
</script>



<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-96166887-2', 'auto');
	
	ga('send', 'pageview');
}
</script>


</body>
</html>

