<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>Over-representation analysis 3: GOseq with GO term in non-model organisms | </title>
    <link rel="stylesheet" href="/css/style.css" />
    <link rel="stylesheet" href="/css/fonts.css" />
    <link href="//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet">

  </head>

  <body class="page">
    <nav class="header">
      <div class="banner">
<a href="/">

<img src="/post/concept_network2title.png" alt="Daddy&#39;s data analysis for biologist, athletes, and family" />
</a>
</div>

      <div class="head-meta">
      
        <span><a href="/">&larr; Back to Home</a></span>
        <span class="date">2018-09-27</span>
        
        
        
          
        
        
        
        
      
      </div>
    </nav>

<div class="container">
<article>
<div class="article-meta">

  <div class="categories">
  
    <a href="/categories/data-analysis">data-analysis</a>
  
  </div>

  <h1><span class="title">Over-representation analysis 3: GOseq with GO term in non-model organisms</span></h1>

  
  <h3 class="author">Kazu Nozue
</h3>
  

  
  <p>Tags: <a href="/tags/r">R</a>; <a href="/tags/reproducibility">reproducibility</a>
  </p>
  
  

</div>



<main>
<div id="if-you-want-to-analyze-non-model-organisms-what-you-an-do" class="section level1">
<h1>If you want to analyze non-model organisms, what you an do?</h1>
<p>Please prepare</p>
<ol class="example" style="list-style-type: decimal">
<li>fasta file for cDNA of your species,</li>
<li>fasta file for cDNA of a close model organisms to your species,</li>
<li>blastn program installed in your computer (UNIX (or Mac OSX) or LINUX) (<a href="https://www.ncbi.nlm.nih.gov/books/NBK52640/">detailed instruciton from NCBI</a>) <!--[](/post/2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term_files/Trello_example_Screen Shot 2018-09-20 at 9.58.20 AM.png)--> # required libraries</li>
</ol>
<pre class="r"><code>library(tidyverse);library(readr);library(readxl)</code></pre>
</div>
<div id="reciprocal-blastn" class="section level1">
<h1>Reciprocal BLASTN</h1>
<pre class="r"><code># look Julin&#39;s scripts</code></pre>
</div>
<div id="annotate-br-genes-with-atgoslim" class="section level1">
<h1>Annotate Br genes with Atgoslim</h1>
<pre class="r"><code>  ## making Brgo.v3.0anno.Atgoslim.BP.list (run once)
  ### prerequisit: results from reciprocal BLAST against Arabidopsis thaliana
  Br.v3.0anno.At.BLAST&lt;-read_csv(file=&quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term/Brapa_V3.0_annotated.csv&quot;) # BLASTed by Julin</code></pre>
<pre><code>## Error in read_csv(file = &quot;2018-09-27-over-representation-analysis-3-goseq-with-non-model-go-term/Brapa_V3.0_annotated.csv&quot;): could not find function &quot;read_csv&quot;</code></pre>
<pre class="r"><code>Br.v3.0anno.At.BLAST</code></pre>
<pre><code>## Error in eval(expr, envir, enclos): object &#39;Br.v3.0anno.At.BLAST&#39; not found</code></pre>
</div>
<div id="section" class="section level1">
<h1></h1>
<pre class="r"><code>  # asign At GO into corresponding Br genes
  Brgo.v3.0anno.Atgoslim.BP.list&lt;-list()
  ###### under construction #############
  
  for(i in 1:length(Br.v2.5anno.At.BLAST$name)) {
    if(is.null(Atgoslim.TAIR.BP.list[[as_vector(Br.v2.5anno.At.BLAST[i,&quot;AGI&quot;])]])) next else {
      Brgo.v2.5anno.Atgoslim.BP.list[[i]]&lt;-Atgoslim.TAIR.BP.list[[as_vector(Br.v2.5anno.At.BLAST[i,&quot;AGI&quot;])]]
      names(Brgo.v2.5anno.Atgoslim.BP.list)[[i]]&lt;-as_vector(Br.v2.5anno.At.BLAST[i,&quot;name&quot;])
    }
  }
  table(sapply(Brgo.v2.5anno.Atgoslim.BP.list,is.null))
  Brgo.v2.5anno.Atgoslim.BP.list.list&lt;-Brgo.v2.5anno.Atgoslim.BP.list[!sapply(Brgo.v2.5anno.Atgoslim.BP.list,is.null)]
  table(sapply(Brgo.v2.5anno.Atgoslim.BP.list,is.null))
  #save(Brgo.v2.5anno.Atgoslim.BP.list,file=&quot;../../Annotation/input/Brgo.v2.5anno.Atgoslim.BP.list.Rdata&quot;)
  save(Brgo.v2.5anno.Atgoslim.BP.list,file=file.path(&quot;common&quot;,&quot;Annotation&quot;,&quot;output&quot;,&quot;v2.5annotation&quot;,&quot;Brgo.v2.5anno.Atgoslim.BP.list.Rdata&quot;))
#} else {print(&quot;change directory&quot;);stop}</code></pre>
</div>
<div id="uner-construction" class="section level1">
<h1>uner construction</h1>
<pre class="r"><code>load(&quot;../../Annotation/input/Brgo.Atgoslim.BP.list.Rdata&quot;) 
library(ShortRead);library(goseq);library(GO.db);library(&quot;annotate&quot;)
# for ggplot heatmap
#library(WGCNA);library(ggplot2);library(reshape2);library(scales)
# read cDNA fasta file with v3.0 annotation
Br_cdna&lt;-readDNAStringSet(&quot;http://brassicadb.org/brad/datasets/pub/Genomes/Brassica_rapa/V3.0/Brapa_genome_v3.0_cds.fasta.gz&quot;)
head(Br_cdna)</code></pre>
</div>
<div id="goseq-ora-function-for-brassica-rapa" class="section level1">
<h1>GOseq ORA function for Brassica rapa</h1>
<pre class="r"><code>GOseq.Brgo.Atgoslim.BP.list.ORA&lt;-function(genelist,padjust=0.05,ontology=&quot;BP&quot;,custom.category.list=Brgo.Atgoslim.BP.list,cdna=Br_cdna) { # return GO enrichment table, padjus, padjust=0.05. 
  bias&lt;-nchar(cdna)
  names(bias)&lt;-names(cdna)
  TF&lt;-(names(bias) %in% genelist)*1
  names(TF)&lt;-names(bias)
  #print(TF)
  pwf&lt;-nullp(TF,bias.data=bias)
  #print(pwf$DEgenes)
  GO.pval &lt;- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #GO.pval &lt;- goseq(pwf,gene2cat=Brgo.DF3,use_genes_without_cat=TRUE) # format became different in new goseq version (021111)
  
  #head(GO.pval) 
  if(ontology==&quot;BP&quot;) {
    GO.pval2&lt;-subset(GO.pval,ontology==&quot;BP&quot;)
  } else if(ontology==&quot;CC&quot;) {
    GO.pval2&lt;-subset(GO.pval,ontology==&quot;CC&quot;)
  } else {
    GO.pval2&lt;-subset(GO.pval,ontology==&quot;MF&quot;)
  }
    
  GO.pval2$over_represented_padjust&lt;-p.adjust(GO.pval2$over_represented_pvalue,method=&quot;BH&quot;)
  if(GO.pval2$over_represented_padjust[1]&gt;padjust) return(&quot;no enriched GO&quot;)
  else {
    enriched.GO&lt;-GO.pval2[GO.pval2$over_represented_padjust&lt;padjust,] 
    print(&quot;enriched.GO is&quot;)
    print(enriched.GO)
    
    ## write Term and Definition 
    for(i in 1:dim(enriched.GO)[1]) {
      if(is.null(Term(GOTERM[enriched.GO[i,&quot;category&quot;]]))) {next} else {
      enriched.GO$Term[i]&lt;-Term(GOTERM[[enriched.GO[i,&quot;category&quot;]]])
      enriched.GO$Definition[i]&lt;-Definition(GOTERM[[enriched.GO[i,&quot;category&quot;]]])
      }
    }
    return(enriched.GO)
  }
}</code></pre>
</div>

</main>














<nav class="post-nav">
  <span class="nav-prev"><a href="/post/2018/09/27/over-representation-analysis-3-goseq-with-non-model-go-term/">&larr; Over-representation analysis 2: GOseq with custom categories</a></span>
  <span class="nav-next"><a href="/post/2018/09/27/over-representation-analysis-4-heatmap-visualization/">Over-representation analysis 4:  heatmap visualization &rarr;</a></span>
</nav>



</article>
</div>

<script async src="//yihui.name/js/center-img.js"></script>

<footer>

<div class="footer">
  <ul class="menu">
    
    <li><a href="/"><span data-hover="Home">Home</span></a></li>
    
    <li><a href="/about/"><span data-hover="About">About</span></a></li>
    
    <li><a href="/categories/"><span data-hover="Categories">Categories</span></a></li>
    
    <li><a href="/tags/"><span data-hover="Tags">Tags</span></a></li>
    
    <li><a href="/post/"><span data-hover="Post">Post</span></a></li>
    
    <li><a href="/index.xml"><span data-hover="Subscribe RSS feed">Subscribe RSS feed</span></a></li>
    
  </ul>
  
</div>
</footer>


<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script>
hljs.configure({languages: []});
hljs.initHighlightingOnLoad();
</script>



<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-96166887-2', 'auto');
	
	ga('send', 'pageview');
}
</script>


</body>
</html>

),i] %&gt;% as_vector()
  temp&lt;-data.frame(AGI=genes,category=names(hormone.responsiveness6.DF.s.list2.DF2)[i])
  #At.gene.name %&gt;% left_join(temp,by=c(&quot;name&quot;=&quot;AGI&quot;)) -&gt; At.gene.name
At.gene.name %&gt;% left_join(temp,by=&quot;AGI&quot;) -&gt; At.gene.name 
  }
names(At.gene.name)[5:24] &lt;- names(hormone.responsiveness6.DF.s.list2.DF2)
At.gene.name %&gt;% filter(str_detect(AGI,pattern=&quot;AT1G|AT2G|AT3G|AT4G|AT5G|ATC|ATM&quot;)) %&gt;% unite(category,5:24,sep=&quot;,&quot;)-&gt;test2
test2[1:10,] %&gt;%  mutate(category=gsub(&quot;NA&quot;,&quot;&quot;,category)) %&gt;% mutate(category=gsub(&quot;,&quot;,&quot;&quot;,category))
# only select genes with AGI name and concatenate categories
At.gene.name %&gt;% filter(str_detect(AGI,pattern=&quot;AT1G|AT2G|AT3G|AT4G|AT5G|ATC|ATM&quot;)) %&gt;% unite(category,5:24,sep=&quot;,&quot;) %&gt;%  mutate(category=gsub(&quot;NA,&quot;,&quot;&quot;,category)) %&gt;% mutate(category=gsub(&quot;,NA&quot;,&quot;&quot;,category)) %&gt;% mutate(category=gsub(&quot;NA&quot;,&quot;&quot;,category)) -&gt;test3
# convert into list object
temp.list&lt;-list()
for(i in 1:dim(test3)[1]) {
  temp.list[[i]]&lt;-test3 %&gt;% slice(i) %&gt;% pull(category)
}
names(temp.list)&lt;-test3 %&gt;% pull(AGI)
# split concatanated categories in each gene
At.hormone.responsive.list&lt;-lapply(temp.list,function(x) unlist(strsplit(x, split=&quot;,&quot;)))
# save
save(At.hormone.responsive.list,file=&quot;2018-09-27-over-representation-analysis-2-goseq-with-non-model-go-term_files/At.hormone.responsive.list.Rdata&quot;)</code></pre>
</div>
<div id="goseq-ora-function-with-arabidopss-thaliana-hormone-responsive-genes" class="section level1">
<h1>GOseq ORA function with Arabidopss thaliana hormone responsive genes</h1>
<pre class="r"><code>load(&quot;2018-09-27-over-representation-analysis-2-goseq-with-non-model-go-term_files/At.hormone.responsive.list.Rdata&quot;)
head(At.hormone.responsive.list)</code></pre>
<pre><code>## $AT1G50920
## character(0)
## 
## $AT1G36960
## character(0)
## 
## $AT1G44020
## character(0)
## 
## $AT1G15970
## character(0)
## 
## $AT1G73440
## character(0)
## 
## $AT1G75120
## character(0)</code></pre>
<pre class="r"><code>library(ShortRead);library(goseq);library(GO.db);library(&quot;annotate&quot;)

#  bias.data vector must have the same length as DEgenes vector!
GOseq.At.customcat.ORA&lt;-function(genelist,padjust=0.05,custom.category.list=At.hormone.responsive.list,cdna=At_cdna) { # return GO enrichment table, padjus, padjust=0.05. New BLAST2GO results with only BP Brgo.v2.5.BP.list is used (042518). Brgo.v2.5.BP.list is based on Hajar&#39;s Blast2go, which should be replaced with Brgo.Atgoslim.BP.list and giving a new name to this function (080818). cdna is DNAstring object.
  #bias&lt;-nchar(cdna)
  bias&lt;-Biostrings::width(cdna)
  # add name to &quot;bias&quot;. cdna
  #names(bias)&lt;-names(cdna)
 names(bias) &lt;- tibble(name=names(cdna)) %&gt;% separate(name,into=c(&quot;name2&quot;,&quot;Symbol&quot;,&quot;description&quot;),sep=&quot; \\|&quot;,extra=&quot;drop&quot;,fill=&quot;left&quot;) %&gt;%  mutate(AGI=str_remove(name2,pattern=&quot;\\.[[:digit:]]+&quot;)) %&gt;% pull(AGI)
  TF&lt;-(names(bias) %in% genelist)*1
  names(TF)&lt;-names(bias)
  #print(TF)
  pwf&lt;-nullp(TF,bias.data=bias)
  #print(pwf$DEgenes)
  GO.pval &lt;- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #GO.pval &lt;- goseq(pwf,gene2cat=Brgo.DF3,use_genes_without_cat=TRUE) # format became different in new goseq version (021111)
  
  # #head(GO.pval) 
  # if(ontology==&quot;BP&quot;) {
  #   GO.pval2&lt;-subset(GO.pval,ontology==&quot;BP&quot;)
  # } else if(ontology==&quot;CC&quot;) {
  #   GO.pval2&lt;-subset(GO.pval,ontology==&quot;CC&quot;)
  # } else {
  #   GO.pval2&lt;-subset(GO.pval,ontology==&quot;MF&quot;)
  # }
    
  GO.pval$over_represented_padjust&lt;-p.adjust(GO.pval$over_represented_pvalue,method=&quot;BH&quot;)
  if(GO.pval$over_represented_padjust[1]&gt;padjust) return(&quot;no enriched GO&quot;)
  else {
    enriched.GO&lt;-GO.pval[GO.pval$over_represented_padjust&lt;padjust,] 
    print(&quot;enriched.GO is&quot;)
    print(enriched.GO)
    
    # ## write Term and Definition 
    # for(i in 1:dim(enriched.GO)[1]) {
    #   if(is.null(Term(GOTERM[enriched.GO[i,&quot;category&quot;]]))) {next} else {
    #   enriched.GO$Term[i]&lt;-Term(GOTERM[[enriched.GO[i,&quot;category&quot;]]])
    #   enriched.GO$Definition[i]&lt;-Definition(GOTERM[[enriched.GO[i,&quot;category&quot;]]])
    #   }
    #}
    return(enriched.GO)
  }
}</code></pre>
</div>
<div id="test-the-function" class="section level1">
<h1>test the function</h1>
<pre class="r"><code>GOseq.At.customcat.ORA(genelist=hormone.responsiveness6.DF.s.list2.DF2$IAAup)</code></pre>
<pre><code>## Using manually entered categories.</code></pre>
<pre><code>## Calculating the p-values...</code></pre>
<p><img src="/post/2018-09-27-over-representation-analysis-2-goseq-with-non-model-go-term_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre><code>## [1] &quot;enriched.GO is&quot;
##      category over_represented_pvalue under_represented_pvalue numDEInCat
## 11      IAAup            0.000000e+00                1.0000000        198
## 12     MJdown            1.514835e-19                1.0000000         23
## 5        BLup            6.983620e-17                1.0000000         11
## 19 WRKY33down            1.037009e-14                1.0000000         18
## 16  PIFtarget            1.176369e-13                1.0000000         11
## 3       ACCup            1.326587e-10                1.0000000          8
## 13       MJup            5.101117e-08                1.0000000         18
## 1     ABAdown            5.084168e-07                0.9999999         12
## 4      BLdown            3.667101e-06                0.9999999          5
## 8      GAdown            3.245315e-05                0.9999983          5
## 18       SAup            4.149904e-05                0.9999947          8
## 7        CKup            9.865021e-04                0.9999212          4
## 10    IAAdown            1.006668e-02                0.9989673          3
##    numInCat over_represented_padjust
## 11      198             0.000000e+00
## 12      230             1.514835e-18
## 5        28             4.655747e-16
## 19      195             5.185045e-14
## 16       52             4.705474e-13
## 3        34             4.421957e-10
## 13      519             1.457462e-07
## 1       269             1.271042e-06
## 4        33             8.149112e-06
## 8        50             6.490630e-05
## 18      168             7.545280e-05
## 7        60             1.644170e-03
## 10       61             1.548720e-02</code></pre>
<pre><code>##      category over_represented_pvalue under_represented_pvalue numDEInCat
## 11      IAAup            0.000000e+00                1.0000000        198
## 12     MJdown            1.514835e-19                1.0000000         23
## 5        BLup            6.983620e-17                1.0000000         11
## 19 WRKY33down            1.037009e-14                1.0000000         18
## 16  PIFtarget            1.176369e-13                1.0000000         11
## 3       ACCup            1.326587e-10                1.0000000          8
## 13       MJup            5.101117e-08                1.0000000         18
## 1     ABAdown            5.084168e-07                0.9999999         12
## 4      BLdown            3.667101e-06                0.9999999          5
## 8      GAdown            3.245315e-05                0.9999983          5
## 18       SAup            4.149904e-05                0.9999947          8
## 7        CKup            9.865021e-04                0.9999212          4
## 10    IAAdown            1.006668e-02                0.9989673          3
##    numInCat over_represented_padjust
## 11      198             0.000000e+00
## 12      230             1.514835e-18
## 5        28             4.655747e-16
## 19      195             5.185045e-14
## 16       52             4.705474e-13
## 3        34             4.421957e-10
## 13      519             1.457462e-07
## 1       269             1.271042e-06
## 4        33             8.149112e-06
## 8        50             6.490630e-05
## 18      168             7.545280e-05
## 7        60             1.644170e-03
## 10       61             1.548720e-02</code></pre>
</div>

</main>














<nav class="post-nav">
  <span class="nav-prev"><a href="/post/2018/10/01/differential-expression-analysis-with-public-available-sequencing-data/">&larr; Differential expression analysis with public available sequencing data</a></span>
  <span class="nav-next"><a href="/post/2018/09/27/over-representation-analysis-3-goseq-with-non-model-go-term/">Over-representation analysis 3: GOseq with GO term in non-model organisms &rarr;</a></span>
</nav>



</article>
</div>

<script async src="//yihui.name/js/center-img.js"></script>

<footer>

<div class="footer">
  <ul class="menu">
    
    <li><a href="/"><span data-hover="Home">Home</span></a></li>
    
    <li><a href="/about/"><span data-hover="About">About</span></a></li>
    
    <li><a href="/categories/"><span data-hover="Categories">Categories</span></a></li>
    
    <li><a href="/tags/"><span data-hover="Tags">Tags</span></a></li>
    
    <li><a href="/post/"><span data-hover="Post">Post</span></a></li>
    
    <li><a href="/index.xml"><span data-hover="Subscribe RSS feed">Subscribe RSS feed</span></a></li>
    
  </ul>
  
</div>
</footer>


<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script>
hljs.configure({languages: []});
hljs.initHighlightingOnLoad();
</script>



<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-96166887-2', 'auto');
	
	ga('send', 'pageview');
}
</script>


</body>
</html>

